django/template/loaders/cached.py

```diff
@@ -82,7 +82,7 @@ def cache_key(self, template_name, template_dirs, skip=None):
         return ("%s-%s-%s" % (template_name, skip_prefix, dirs_prefix)).strip('-')
 
     def generate_hash(self, values):
-        return hashlib.sha1(force_bytes('|'.join(values))).hexdigest()
```

Comment:
I don't think we can "just" change this to sha256. Users who upgrade will suddenly, easily see a spike in CPU load, potentially taking down their site. I think we need to be more careful about the upgrade here.